### DISCUSSION TITLE
base.Compressor.get_filepath hexdigest part optional with basename passed?

### DISCUSSION THREAD

**Discussion by berycz** [django-compressor/django-compressor#1209] [Ideas]
Hi,
I'm not using compressor yet, so I'm not that familiar with it, but I'm considering it.
But I use whitenoise, so I was thinking if the `hexdigest` part of filenames must be mandatory when the `basename` is passed? Since whitenoise takes care of versioning already.
Or is there any other reason for it?
I would use it mainly for sass->css compilation.
Would it make sense to have a setting for it, maybe `COMPRESS_FILENAME_ADD_HASH_PART` / `COMPRESS_BASENAME_ONLY` or something?

**Comment by carltongibson** [comment]
The Whitenoise docs have this about using with compressor: 

https://whitenoise.readthedocs.io/en/latest/django.html#django-compressor

So I guess I'd say, set up for offline compression (i.e. running the `compress` command) and then see how it goes. 

I'm not aware of any issues...

**Reply by berycz** [reply]
that's not what I'm asking :)
my question is, if I have something like this
```html
{% compress css main %}
<link rel="stylesheet" href="/static/css/one.css" type="text/css" charset="utf-8">
<style type="text/css">p { border:5px solid green;}</style>
<link rel="stylesheet" href="/static/css/two.css" type="text/css" charset="utf-8">
{% endcompress %}
```
I get a `main.f7c661b7a124.css` file, where the `f7c661b7a124` hash is for versioning I guess?
But `whitenoise` takes care of versioning for me when I do `collectstatic`, so I would prefer to have the filename only `main.css` when I specify the `block_name` and thus `get_filepath` receives the `basename`

here https://github.com/django-compressor/django-compressor/blob/develop/compressor/base.py#L146 to have something like
```py
        # parts.extend([get_hexdigest(content, 12), self.resource_kind])  # current code
        if basename is None or not settings.BASENAME_ONLY:  # my idea
            parts.append(get_hexdigest(content, 12))
        parts.append(self.resource_kind)
```

**Reply by carltongibson** [reply]
You might be able to do something like that… 😬 In part compressor uses that to work out if the file has already been processed, so I'm not sure making it pluggable would fly 🤔

**Reply by berycz** [reply]
aha, I assumed that's based only on mtime of the files, hmm
I guess I will have to install it, try it out and read the code more thoroughly

thanks for now :)

**Reply by carltongibson** [reply]
No problem, not sure how helpful it was. 

Have a play. See if you can get you want going.
